### PR TITLE
issue: 863457 Change logic of interface selection in attach receiver

### PR DESCRIPTION
**Pull request by igor-ivanov**
Set high priority for bindtodevice rule.

Signed-off-by: Igor Ivanov igor.ivanov.va@gmail.com

@alexvm @OphirMunk please look at this change.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/491/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/506/ for details (Mellanox internal link).

**Comment by rosenbaumalex** [comment]
Suggest to:

1. Change '?' to 'if()' statement and log if bound due to bindtodevice or simple bind() call for more details. 

2. Keep addr.to_str() logging for port info. str will be 0.0.0.0:5000

**Comment by igor-ivanov** [comment]
updated

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/492/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/507/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@OphirMunk it is ready for merge

**Comment by ilansmith** [comment]
@igor-ivanov - Could you please rebase and update your repository so this gets bot tested again? I'd like to merge it.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/664/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/656/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@OphirMunk @ilansmith is it ready to be included in vNext?

**Comment by igor-ivanov** [comment]
@OphirMunk @NirNitzani Do you see any issues with this bug fix preventing these changes to be included in master?

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1550/ for details (Mellanox internal link).

File: src/vma/sock/sockinfo.cpp [hunk 1]
```diff
@@ -967,6 +975,8 @@ bool sockinfo::attach_as_uc_receiver(role_t role, bool skip_rules /* = false */)
 			}
 		}
 	}
+	si_logdbg("Attached to specific local if: (%d.%d.%d.%d) addr: %s", NIPQUAD(local_if), addr.to_str());
```
Comment (liranoz12) [inline src/vma/sock/sockinfo.cpp]: The log messages are better without the change.
While _local_if_ == INADDR_ANY, we will call _attach_receiver()_ for each offloaded ip but we will only print the information of the last ip
what do you think ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1581/ for details (Mellanox internal link).

**Comment by ilansmith** [comment]
Merged #210.

**Pull request closed without merging**